### ISSUE TITLE
[Feature] Proxy support

### ISSUE THREAD

**Issue by mhf-ir** [ifl0w/RandomWallpaperGnome3#90]
Enable proxy support better to be socks, or even http different from gnome network manager.

**Comment by ifl0w** [comment]
With the release of version 2.7.0 the extension should use the system-wide proxy settings.

I assume your request is to have a separate proxy setting in the UI which should be used when requesting wallpapers?
Personally, I do not see a good reason to implement this feature. For now, I'll tag this issue as `wontfix` and leave it open for further discussion.

**Comment by TechyCSR** [comment]
@ifl0w Proxy feature is much required.

**Comment by ifl0w** [comment]
Hey @TechyCSR, what exactly is required here? It doesn't really make sense in my opinion to integrate network specific features in this extension. Also any system wide proxy or tunnel should apply to all requests anyways.

**Comment by TechyCSR** [comment]
@ifl0w yeah you are right in your way but some of the domains are blocked in specific location which don't allows us to access those wallpaper/images or Requests .
So to overcome this we can have a proxy .

**Comment by libook** [comment]
At the system level, I might use a proxy for privacy protection; however, when accessing specific wallpaper resources, I might need to use a separate proxy for a private network (such as my home NAS from outside).

So it may be cool to have a proxy input for each wallpaper source.

**Comment by Rondom** [comment]
I think it is better to solve this issue on the system level rather than inside a random wallpaper extension. Say you want to access your wallpapers in another program, how would you go about that? Would you require a special feature in that program as well?
Depending on the particular setup, it can probably solved by configuring exceptions for certain hosts or specifying more complex rules via an own proxy-pac-file.
